pkg/agent/openflow/pipeline.go

```diff
@@ -1906,15 +2065,27 @@ func (c *client) loadBalancerServiceFromOutsideFlow(svcIP net.IP, svcPort uint16
 
 // serviceLearnFlow generates the flow with learn action which adds new flows in
 // sessionAffinityTable according to the Endpoint selection decision.
-func (c *client) serviceLearnFlow(groupID binding.GroupIDType, svcIP net.IP, svcPort uint16, protocol binding.Protocol, affinityTimeout uint16) binding.Flow {
+func (c *client) serviceLearnFlow(groupID binding.GroupIDType, svcIP net.IP, svcPort uint16, protocol binding.Protocol, affinityTimeout uint16, nodeLocalExternal bool) binding.Flow {
 	// Using unique cookie ID here to avoid learned flow cascade deletion.
 	cookieID := c.cookieAllocator.RequestWithObjectID(cookie.Service, uint32(groupID)).Raw()
-	learnFlowBuilder := c.pipeline[serviceLBTable].BuildFlow(priorityLow).
-		MatchRegRange(int(serviceLearnReg), marksRegServiceNeedLearn, serviceLearnRegRange).
-		MatchDstIP(svcIP).
-		MatchProtocol(protocol).
-		MatchDstPort(svcPort, nil).
-		Cookie(cookieID)
+
+	var learnFlowBuilder binding.FlowBuilder
+	if !nodeLocalExternal {
+		learnFlowBuilder = c.pipeline[serviceLBTable].BuildFlow(priorityLow).
```

Comment:
Add comments to explain this flow and why it is needed when nodeLocalExternal is false.

Reply:
Reminder.

Reply:
Done